src/util/KinBodyParser.cpp

```diff
@@ -0,0 +1,532 @@
+{
+  // `type` attribute should be one of `none`, `box`, `sphere`, `trimesh`,
+  // or `cylinder`
+  auto typeAttr = dart::utils::getAttributeString(geomEle, "type");
+
+  dart::dynamics::ShapeNode* shapeNode = nullptr;
+  std::string fileNameOfCollision;
+  double uniScaleOfCollision = 1.0;
+
+  auto shapeNodeName = resolveShapeName(geomEle, bodyNode);
+
+  if (typeAttr == "none")
+  {
+    // Do nothing.
+  }
+  else if (typeAttr == "box")
+  {
+    auto extents = dart::utils::getValueVector3d(geomEle, "extents");
+    auto shape = std::make_shared<dart::dynamics::BoxShape>(extents);
+
+    shapeNode = bodyNode->createShapeNodeWith<
+        dart::dynamics::CollisionAspect,
+        dart::dynamics::DynamicsAspect>(shape, shapeNodeName);
+  }
+  else if (typeAttr == "sphere")
+  {
+    auto radius = dart::utils::getValueDouble(geomEle, "radius");
+    auto shape = std::make_shared<dart::dynamics::SphereShape>(radius);
+
+    shapeNode = bodyNode->createShapeNodeWith<
+        dart::dynamics::CollisionAspect,
+        dart::dynamics::DynamicsAspect>(shape, shapeNodeName);
+  }
+  else if (typeAttr == "cylinder")
+  {
+    auto radius = dart::utils::getValueDouble(geomEle, "radius");
+    auto height = dart::utils::getValueDouble(geomEle, "height");
+    auto shape
+        = std::make_shared<dart::dynamics::CylinderShape>(radius, height);
+
+    shapeNode = bodyNode->createShapeNodeWith<
+        dart::dynamics::CollisionAspect,
+        dart::dynamics::DynamicsAspect>(shape, shapeNodeName);
+  }
+  else if(typeAttr == "trimesh")
+  {
+    std::string collisionOrData;
+
+    auto hasCollisionEle = dart::utils::hasElement(geomEle, "collision");
+    auto hasDataEle = dart::utils::hasElement(geomEle, "data");
+
+    if (hasCollisionEle && hasDataEle)
+    {
+      dtwarn << "[KinBodyParser]: <Geom> element contains both of <Collision> "
+             << "and <Data> as the child elements. <Data> element is "
+             << "ignored.\n";
+    }
+
+    if (hasCollisionEle)
+    {
+      collisionOrData = "collision";
+    }
+    else if (hasDataEle)
+    {
+      collisionOrData = "data";
+    }
+    else
+    {
+      dterr << "[KinBodyParser] <Geom> element doesn't contain neither of "
+            << "<Collision> and <Data> as the child element. Ignoring this "
+            << "<Geom>.\n";
+      return;
+    }
+
+    auto fileNameAndScale = resolveFileNameAndScale(geomEle, collisionOrData);
+    fileNameOfCollision = fileNameAndScale.first;
+    uniScaleOfCollision = fileNameAndScale.second;
+    auto shape = readMeshShape(
+        fileNameOfCollision, uniScaleOfCollision, baseUri, retriever);
+
+    shapeNode = bodyNode->createShapeNodeWith<
+        dart::dynamics::CollisionAspect,
+        dart::dynamics::DynamicsAspect>(shape, shapeNodeName);
+  }
+  else
+  {
+    dterr << "[KinBodyParser] Attempts to parse unsupported geom type '"
+          << typeAttr << "'. Ignoring this <Geom>.\n";
+    return;
```

Comment:
I thought we don't want to create visual geometry for invalid geom type but changed not to so (create visual geometry) because, technically, we still can create one as long as `<Render>` element is provided.